src/config.rs

```diff
@@ -521,8 +521,54 @@ pub(crate) struct RenderedLinkConfig {
 #[serde(rename_all = "kebab-case")]
 #[serde(deny_unknown_fields)]
 pub(crate) struct IssueLinksConfig {
-    #[serde(default = "default_true")]
-    pub(crate) check_commits: bool,
+    #[serde(default)]
+    pub(crate) check_commits: IssueLinksCheckCommitsConfig,
+}
+
+#[derive(PartialEq, Eq, Debug)]
+pub(crate) enum IssueLinksCheckCommitsConfig {
+    Activated(bool),
```

Comment:
Maybe `Off` / `Uncanonicalized` / `All` would be a bit easier to understand? It took me a while to grok what this enum means. But not a big deal.

Reply:
Adjusted the code as suggested, and added some comments to improve the meaning.